File: apps/crm-api/app/api/routes/tasks.py
"""Tasks — STUDENT EXERCISE (skeleton).

Implement CRUD mirroring app/api/routes/customers.py.
Model fields (match the crm-web `Task` type): title, due_date, status
("todo"|"in_progress"|"done"), priority ("low"|"medium"|"high"), assignee.
See app/api/routes/contacts.py for the step-by-step checklist.
"""

from fastapi import APIRouter, HTTPException, status

from app.api.deps import CurrentUser  # noqa: F401

router = APIRouter(prefix="/tasks", tags=["tasks"])


@router.get("")
def list_tasks() -> list[dict]:
    raise HTTPException(
        status_code=status.HTTP_501_NOT_IMPLEMENTED,
        detail="Tasks endpoint not implemented yet — student exercise.",
    )

File: apps/crm-web/src/data/mock/tasks.ts
import { TaskPriority, TaskStatus } from "@/app/(dashboard)/tasks/enums";
import type { Task } from "@/app/(dashboard)/tasks/types";

export const tasks: Task[] = [
  {
    id: 1,
    title: "Gọi điện theo dõi với Acme Corp",
    due_date: "2026-06-16",
    status: TaskStatus.TODO,
    priority: TaskPriority.HIGH,
    assignee: "Mai Anh",
  },
  {
    id: 2,
    title: "Gửi báo giá cho Globex",
    due_date: "2026-06-14",
    status: TaskStatus.IN_PROGRESS,
    priority: TaskPriority.HIGH,
    assignee: "Quốc Bảo",
  },
  {
    id: 3,
    title: "Chuẩn bị demo cho Initech",
    due_date: "2026-06-20",
    status: TaskStatus.TODO,
    priority: TaskPriority.MEDIUM,
    assignee: "Mai Anh",
  },
  {
    id: 4,
    title: "Cập nhật hồ sơ khách hàng Hooli",
    due_date: "2026-06-12",
    status: TaskStatus.DONE,
    priority: TaskPriority.LOW,
    assignee: "Quốc Bảo",
  },
];
